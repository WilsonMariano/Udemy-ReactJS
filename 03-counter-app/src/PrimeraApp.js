import React from 'react';

const PrimeraApp = ({ saludo, subtitulo = "Soy un subtÃ­tulo" }) => {

    return (
        <>
            <h1>{ saludo }</h1>
            <p>{ subtitulo }</p>
        </>
    );
}

export default PrimeraApp;