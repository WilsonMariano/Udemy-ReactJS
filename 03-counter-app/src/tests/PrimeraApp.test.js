import '@testing-library/jest-dom';
import React from 'react';
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';

describe('Pruebas en <PrimeraApp />', () => {

    // test('debe mostrar el mensaje "Hola, soy Mariano"', () => {
    //     const saludo = 'Hola, soy Mariano';
    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // });

    test('debe mostrar <PrimeraApp /> correctamennte', () => {

        const saludo = "Hola, soy Mariano";
        const wrapper = shallow(<PrimeraApp saludo={ saludo } />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar el subtÃ­tulo enviado por props', () => {

        const saludo = "Hola, soy Mariano";
        const subtitulo = "Soy un programador";

        const wrapper = shallow(
            <PrimeraApp 
                saludo={ saludo }
                subtitulo={ subtitulo } 
            />
        );

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subtitulo);
    });
});