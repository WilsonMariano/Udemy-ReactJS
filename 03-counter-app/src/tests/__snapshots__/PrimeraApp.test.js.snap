// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe mostrar <PrimeraApp /> correctamennte 1`] = `
<Fragment>
  <h1>
    Hola, soy Mariano
  </h1>
  <p>
    Soy un subt√≠tulo
  </p>
</Fragment>
`;
